server/shared/storage/amazon.ts

```diff
@@ -0,0 +1,123 @@
+import IStorage, {
+  ContentResponse,
+  DeleteResponse,
+  FileListResponse
+} from './interface';
+import { AWSError } from 'aws-sdk';
+import { Config } from '../../config';
+import miss from 'mississippi';
+import path from 'path';
+import S3 from 'aws-sdk/clients/s3';
+
+const isNotFound = (err: AWSError): boolean => err.code === 'NoSuchKey';
+
+class Amazon implements IStorage {
+  #bucket: string;
+  #client: S3;
+
+  constructor(config: Config) {
+    const amazonConfig = config.storage.amazon;
+
+    const s3Config = {
+      accessKeyId: amazonConfig.key,
+      secretAccessKey: amazonConfig.secret,
+      region: amazonConfig.region,
+      signatureVersion: 'v4',
+      apiVersion: '2006-03-01',
+      maxRetries: 3
+    };
+
+    this.#bucket = amazonConfig.bucket;
+    this.#client = new S3(s3Config);
+  }
```

Comment:
Consider whether this is simpler to read/less indirection:
```suggestion
  constructor(config: Config) {
    this.#bucket = config.storage.amazon.bucket;
    this.#client = new S3({
      accessKeyId: config.storage.amazon.key,
      secretAccessKey: config.storage.amazon.secret,
      region: config.storage.amazon.region,
      signatureVersion: 'v4',
      apiVersion: '2006-03-01',
      maxRetries: 3
    });
  }
```
or some similar variation.
Basically `config.storage.amazon` is S3 config so having `amazonConfig` and `s3Config` seems misleading.